src/local_scheduler/local_scheduler.cc

```diff
@@ -622,77 +637,64 @@ void process_message(event_loop *loop,
     }
 
   } break;
-  case TASK_DONE: {
+  case MessageType_TaskDone: {
   } break;
-  case EVENT_LOG_MESSAGE: {
-    /* Parse the message. TODO(rkn): Redo this using flatbuffers to serialize
-     * the message. */
-    uint8_t *message = (uint8_t *) utarray_front(state->input_buffer);
-    int64_t offset = 0;
-    int64_t key_length;
-    memcpy(&key_length, &message[offset], sizeof(key_length));
-    offset += sizeof(key_length);
-    int64_t value_length;
-    memcpy(&value_length, &message[offset], sizeof(value_length));
-    offset += sizeof(value_length);
-    uint8_t *key = (uint8_t *) malloc(key_length);
-    memcpy(key, &message[offset], key_length);
-    offset += key_length;
-    uint8_t *value = (uint8_t *) malloc(value_length);
-    memcpy(value, &message[offset], value_length);
-    offset += value_length;
-    CHECK(offset == length);
+  case MessageType_EventLogMessage: {
+    /* Parse the message. */
+    auto message = flatbuffers::GetRoot<EventLogMessage>(
+        utarray_front(state->input_buffer));
     if (state->db != NULL) {
-      RayLogger_log_event(state->db, key, key_length, value, value_length);
+      RayLogger_log_event(
+          state->db, (uint8_t *) message->key()->data(), message->key()->size(),
+          (uint8_t *) message->value()->data(), message->value()->size());
     }
-    free(key);
-    free(value);
   } break;
-  case REGISTER_WORKER_INFO: {
+  case MessageType_RegisterWorkerInfo: {
     /* Update the actor mapping with the actor ID of the worker (if an actor is
      * running on the worker). */
-    register_worker_info *info =
-        (register_worker_info *) utarray_front(state->input_buffer);
-    if (!ActorID_equal(info->actor_id, NIL_ACTOR_ID)) {
+    auto message = flatbuffers::GetRoot<RegisterWorkerInfo>(
+        utarray_front(state->input_buffer));
+    int64_t worker_pid = message->worker_pid();
```

Comment:
you use `int64_t` here, but in the flatbuffer spec it is `ulong`.

We should make them consistent.